WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -1911,6 +1963,130 @@ static GSourceFuncs _handlerIntervention =
+                    fflush(stdout);
+                    return;
+                }
+                static const long PageSize = sysconf(_SC_PAGE_SIZE);
+                unsigned long rssPageNum = std::stoul(items[1]);
+                inBytes = rssPageNum * PageSize;
+                return;
+            };
+
+            auto readStatmLine = [&] (pid_t pid, std::string &statmLine) -> bool
+            {
+                if (pid <= 1)
+                {
+                    printf("Cannot get stats for process id = %u\n", pid);
+                    fflush(stdout);
+                    return false;
+                }
+                std::string procPath = std::string("/proc/") + std::to_string(pid) + "/statm";
+                std::ifstream statmStream(procPath);
+                if (!statmStream.is_open() || !std::getline(statmStream, statmLine))
+                {
+                    printf("Cannot read process 'statm' file for process id = %u\n", pid);
+                    fflush(stdout);
+                    return false;
+                }
+                return true;
+            };
+
+            auto addProcessInfo = [&](JsonObject &metrics)
+            {
+
+                pid_t webprocessPID = GetWebGetProcessIdentifier();
+                uint32_t rssInBytes = 0;
+                std::string statmLine;
+                if (readStatmLine(webprocessPID, statmLine))
+                {
+                    parseRssFromStatmLine(statmLine, rssInBytes);
+                }
+
+                metrics.Set("ProcessRSS", std::to_string(rssInBytes));
+                metrics.Set("ProcessPID", std::to_string(webprocessPID));
+                metrics.Set("AppName", _URL.c_str());
+                metrics.Set("webProcessStatmLine", statmLine);
+            };
+
+            JsonObject metrics;
+            metrics.Set("LaunchState", getProcessLaunchStateString());
+            metrics.Set("AppType", "Web");
+
+            addSystemInfo(metrics);
+            addProcessInfo(metrics);
+
+            gint64 idleTime = 0;
+            if (_idleStart > 0) {
+                idleTime = (g_get_monotonic_time() - _idleStart) / G_USEC_PER_SEC;
+                _idleStart = -1;
+            }
+            metrics.Set("webProcessIdleTime", std::to_string(idleTime));
+
+            std::swap(_launchMetrics, metrics);
+        }
+
+        void collectMetricsOnLoadEnd()
+        {
+            if (_didSendLaunchMetrics)
+                return;
+
+            gint64 pageLoadTimeMs = (g_get_monotonic_time() - _pageLoadStart) / 1000;
+            _launchMetrics.Set("LaunchTime", std::to_string(pageLoadTimeMs));
+            _launchMetrics.Set("AppLoadSuccess", std::to_string(!_loadFailed));
+            _launchMetrics.Set("webPageLoadNum", std::to_string(_pageLoadNum));
+
+            std::string str;
+            _launchMetrics.ToString(str);
+            std::cout<<str<<std::endl;
```

Comment:
Please use notification log as you had it previously

Reply:
I modified to SYSLOG notification.